src/components/Signup.jsx

```diff
@@ -0,0 +1,94 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { Container, Modal, Form, Input, Button } from "semantic-ui-react";
+
+const Signup = () => {
+  const [userEmail, setUserEmail] = useState("");
+  const [userPassword, setUserPassword] = useState("");
+  const [userConfirmPassword, setUserConfirmPassword] = useState("");
+  const [open, setOpen] = useState(false);
+  const [validEmail, setValidEmail] = useState(true);
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    axios({
+      method: "post",
+      url: "https://slowfood.heroku.com/api/auth/",
+      params: {
+        email: userEmail,
+        password: userPassword,
+        password_confirmation: userConfirmPassword,
+        confirm_success_url: "placeholder",
+      },
+    }).then(setOpen(true));
+  };
+
+  const validateEmail = (email) => {
+    const re =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    setValidEmail(re.test(String(email).toLowerCase()));
+  };
```

Comment:
I would extract this to a module

Reply:
We have extracted the function to module.